### PR TITLE
80 feature/bot add component

### PR DESCRIPTION
**Pull request by Ilya-Mir**
Add bot api component

### CODE REVIEW

File: src/containers/dashboard/AddBotApi.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  };
+};
+
+
+export default injectIntl(
+  connect(state => ({
+    userId: state.auth.profile._id,
+    exchanges: state.exchanges,
+    apiKeys: state.apiKeys.ownKeys,
+    is2FAEnable: state.auth.profile.mfaEnabled,
```
Comment (rumkin) [inline src/containers/dashboard/AddBotApi.js]: `is2FAEnable ` -> `is2faEnabled`.

File: src/containers/dashboard/ApiBot.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+  renderModeDropdown = () => {
+    const { currentMode} = this.state;
+    return (
+      <div className="mode_select_wrapper">
+        <ExchangeSelect exchanges={['Active Keys', 'Deleted Keys']}
+          exchange={ currentMode }
+          defaultPlaceholder="Current Mode"
+          onChange={ e => {
+            this.setState({currentMode: e});
```
Comment (rumkin) [inline src/containers/dashboard/ApiBot.js]: Current mode became and event.

File: src/containers/dashboard/ApiBot.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    <div className="table_header_wrapper contract_header_wrapper">
+      <span className="table_header">{header}</span>
+    </div>
+  );
+
+  renderModeDropdown = () => {
+    const { currentMode} = this.state;
+    return (
+      <div className="mode_select_wrapper">
+        <ExchangeSelect exchanges={['Active Keys', 'Deleted Keys']}
```
Comment (rumkin) [inline src/containers/dashboard/ApiBot.js]: Use constants and i18n instead of pure strings.

File: src/containers/dashboard/ApiBot.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+import {FormattedMessage, FormattedDate, injectIntl } from 'react-intl';
+import ReactTable from '../../components/SelectableReactTable';
+import Pagination from '../../components/Pagination';
+import ExchangeSelect from '../../components/ExchangeSelect';
+import { Desktop, Mobile } from '../../generic/MediaQuery';
+import {showTwoFactorAuthModal, showConfirmModal} from '../../actions/modal';
+import {fetchBotKeys, deleteBotKeys} from '../../actions/apiKeys';
+import classNames from 'classnames';
+
+
```
Comment (rumkin) [inline src/containers/dashboard/ApiBot.js]: Add constants ACTIVE_KEYS, DELETED_KEYS for tabs.

File: src/containers/dashboard/ApiBot.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+class BotList extends React.Component {
+
+  state = {
+    currentMode: 'Active Keys',
```
Comment (rumkin) [inline src/containers/dashboard/ApiBot.js]: Use `ACTIVE_KEYS` constant.

File: src/containers/dashboard/ApiBot.js [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+          }}
+        />
+      </div>
+    );
+  }
+
+  renderContent() {
+    const { keysList, currentMode } = this.state;
+    const currentData = keysList.filter(listItem => currentMode === DELETED_KEYS.value ?
+      Object.prototype.hasOwnProperty.call(listItem, 'deletedAt') :
```
Comment (rumkin) [inline src/containers/dashboard/ApiBot.js]: Why not just `listItem.hasOwnProperty('deletedAt')` ?

File: src/lang/en.js [hunk 7]
```diff
@@ -4,11 +4,14 @@ const en = {
...
   'message' : 'Message',
   'simpleValue' : '{value}',
+  'secretIs' : 'Secret: {secret}',
   'noSuchProfile' : 'No such profile',
   'noDataAboutCurrentRound' : 'No data about current round',
   'theKeyIsInUse' : 'The key is in use',
   'enterConfirmationCode' : 'Enter confirmation code',
   'saveAndConfirmSecretCode' : 'Save and confirm secret code',
+  'yourBotSecretTitle' : 'Your Bot Secret Code',
+  'saveYourCode' : 'Safely memory your code, you will no longer be able to get it',
```
Comment (rumkin) [inline src/lang/en.js]: `Store the code safely, you will no longer be able to get it.`

File: src/containers/dashboard/ApiBot.js (hunk 6 above)
Comment (Ilya-Mir) [reply src/containers/dashboard/ApiBot.js]: https://eslint.org/docs/rules/no-prototype-builtins

Comment (rumkin) [reply src/containers/dashboard/ApiBot.js]: No prototype builtins is recommended for prototypeless objects. Can keyList item be such object?

File: src/components/ExchangeSelect.js [hunk 8]
```diff
@@ -16,11 +16,13 @@ class ExchangeSelect extends React.Component {
   };
 
   render() {
-    const { exchangesTitleClasses, exchange, defaultPlaceholder } = this.props;
+    const { exchanges, exchangesTitleClasses, exchange, defaultPlaceholder } = this.props;
     return (
       <div className="add_keys_select_wr">
         <div className={`add_keys_select_value upper upper ${exchangesTitleClasses}`}>{
-          exchange ||  defaultPlaceholder
+          exchange && typeof exchanges[0] === 'object' ?
```
Comment (rumkin) [inline src/components/ExchangeSelect.js]: Rewrite this condition into readable form.

File: src/containers/dashboard/ApiBot.js [hunk 9]
```diff
@@ -67,7 +67,7 @@ class BotList extends React.Component {
     return [
```
Comment (rumkin) [inline src/containers/dashboard/ApiBot.js]: Rewrite into readable form:

```javascript
const commonHeaders = [];

if (currentMode !== ACTIVE_KEYS.value) {
   return commonHeaders;
}
else {
  return [
    ...commonHeaders,
    // other headers
  ];
}
```

**Pull request merged**